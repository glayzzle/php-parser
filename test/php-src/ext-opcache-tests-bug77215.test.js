// eslint-disable prettier/prettier
const parser = require("../main");

describe("php-src tests", function () {
  // ext/opcache/tests/bug77215.phpt
  it("Bug #77215: CFG assertion failure on multiple finalizing switch frees in one block", function () {
    expect(parser.parseCode("<?php\nfunction _crash($eValeur) {\n    switch ($eValeur[\"a\"]) {\n        default:\n            switch($eValeur[\"a\"]) {\n                default:\n                    return 2;\n            }\n    }\n}\nvar_dump(_crash([\"a\" => \"b\"]));\n?>")).toMatchSnapshot();
  });
});
