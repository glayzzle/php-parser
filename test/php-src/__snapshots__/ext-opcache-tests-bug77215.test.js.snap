// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`php-src tests Bug #77215: CFG assertion failure on multiple finalizing switch frees in one block 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "attrGroups": Array [],
          "byref": false,
          "flags": 0,
          "kind": "parameter",
          "name": Identifier {
            "kind": "identifier",
            "name": "eValeur",
          },
          "nullable": false,
          "readonly": false,
          "type": null,
          "value": null,
          "variadic": false,
        },
      ],
      "attrGroups": Array [],
      "body": Block {
        "children": Array [
          Switch {
            "body": Block {
              "children": Array [
                Case {
                  "body": Block {
                    "children": Array [
                      Switch {
                        "body": Block {
                          "children": Array [
                            Case {
                              "body": Block {
                                "children": Array [
                                  Return {
                                    "expr": Number {
                                      "kind": "number",
                                      "value": "2",
                                    },
                                    "kind": "return",
                                  },
                                ],
                                "kind": "block",
                              },
                              "kind": "case",
                              "test": null,
                            },
                          ],
                          "kind": "block",
                        },
                        "kind": "switch",
                        "shortForm": false,
                        "test": OffsetLookup {
                          "kind": "offsetlookup",
                          "offset": String {
                            "isDoubleQuote": true,
                            "kind": "string",
                            "raw": "\\"a\\"",
                            "unicode": false,
                            "value": "a",
                          },
                          "what": Variable {
                            "curly": false,
                            "kind": "variable",
                            "name": "eValeur",
                          },
                        },
                      },
                    ],
                    "kind": "block",
                  },
                  "kind": "case",
                  "test": null,
                },
              ],
              "kind": "block",
            },
            "kind": "switch",
            "shortForm": false,
            "test": OffsetLookup {
              "kind": "offsetlookup",
              "offset": String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"a\\"",
                "unicode": false,
                "value": "a",
              },
              "what": Variable {
                "curly": false,
                "kind": "variable",
                "name": "eValeur",
              },
            },
          },
        ],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "_crash",
      },
      "nullable": false,
      "type": null,
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              Array {
                "items": Array [
                  Entry {
                    "byRef": false,
                    "key": String {
                      "isDoubleQuote": true,
                      "kind": "string",
                      "raw": "\\"a\\"",
                      "unicode": false,
                      "value": "a",
                    },
                    "kind": "entry",
                    "unpack": false,
                    "value": String {
                      "isDoubleQuote": true,
                      "kind": "string",
                      "raw": "\\"b\\"",
                      "unicode": false,
                      "value": "b",
                    },
                  },
                ],
                "kind": "array",
                "shortForm": true,
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "_crash",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
